# Lesson 1

## Two sided test

In [ ]:
import math

Let's see how to compute the statistic and the critical value for the statistic.

In [ ]:
sample_mean = 80.94
pop_mean = 85 
pop_std = 11.6 # We are given the population std -> Z-score and Z-table
n = 25
statistic = (sample_mean - pop_mean)/(pop_std/math.sqrt(n))
print("Statistic is: ", statistic)

Statistic is:  -1.750000000000001


We want to asses if the sample mean is significantly different from the population mean. It can be higher or lower. Therefore, $\bar{x}-\mu$ can be positive or negative.

If we take a 95% confidence interval, the significance level will be: 1 - .95 = 0.05

We need to go to the table of normal distribution and find out what is the critical value of the statistic in order to have an area of 0.95.

As the normal distribution is symmetrical and the total area adds 1, the half of the area will be 0.5.

We know that our significance level $\alpha$ is 0.05. As we're dealing with a **two sided test** (the difference between our sample mean and the population mean can be positive or negative), we divide this by 2 and we get: 0.025. 

Then half of the area minus this amount will tell us how much area is left in one side = 0.5 - 0.025 = 0.475. Now we only need to go to the Z-table and to look for this value. Once tihs value is found, we need to look at the value to the lef and in the same row. To this value we will add the value in the header of the intersection. = -1.96 

Now we can compare the absolute value of our statistic value = 1.75 and the critical value. As our value is smaller than the critical value we **accept  H0** 

# Lesson 2

## One sided test

Now we're going to asses if the sample mean is significantly higher than the population mean. Therefore, the difference $\bar{x}-\mu$ is positive. 



Now we don't need to divide by 2 because the difference can only be positive. 

Out statistic hasn't change, what it has changed is the restriction.

In [ ]:
sample_mean = 80.94
pop_mean = 85
pop_std = 11.6
n = 25
statistic = (sample_mean - pop_mean)/(pop_std/math.sqrt(n))
print("Statistic is: ", statistic)

Statistic is:  -1.750000000000001


Let's compute the new area:

* Confidence level = 95%
* Significance level = 0.05
* Area = 0.5 - 0.05 = 0.45

We go to the table and we find the critical Z-score value:

aprox = -1.64

If we take absolute values, our statistic is higher than the critical value. Therefore we're in the bad reagion -> We **reject the Ho** 

## Using the p-value

Another way to see the same is to wonder, how much area will be under the curve given our statistic. 

To know that, we need to go to the table and find out the area for our statistic -1.75, which is = 0.04006. Look fo the intersection of -1.7 and 0.05 -> -1.75. 

0.04006 es samller than the 0.05 (we're on the bad side) -> We **reject H0**

# Lesson 4

When comparing **two samples of different populations** the statistic to compute is:

$$t = \frac{(\bar{x_{1}}-\bar{x_{2}})}{s_{p}\sqrt{\frac{1}{n_{1}}+\frac{1}{n_{2}}}}$$

with:

$$s_{p}=\sqrt{\frac{(n_{1}-1)s_{1}^{2}+(n_{2}-2)s_{2}^{2}}{(n_{1}+n_{2}-2)}}$$

Be aware that again, the difference between the sample means, can be either positive or negative. Therefore, we are going to conduct a **two sided test** but with a different statistic.

In [ ]:
sample_mean1 = 105.5
sample_std1 = 20.1
n1 = 34
sample_mean2 = 90.9
sample_std2 = 12.2
n2 = 29

pooled_sample_std = math.sqrt(((n1-1)*sample_std1**2 + (n2-1)*sample_std2**2)/(n1+n2-2))
statistic = (sample_mean1-sample_mean2)/(pooled_sample_std*math.sqrt((1/n1)+(1/n2)))
print("T Statistic is: ", statistic)

T Statistic is:  3.4101131776909535


Now we need to use the table corresponding to the t-student distribution instead off the normal distribution[look here](https://www.itl.nist.gov/div898/handbook/eda/section3/eda3672.htm)

This table has on the left most column, the degrees of freedom (34 + 29 - 2) = 61. On the header is the confidence level. Then in our case the intersection between 61 and (0.95/2 = .975) is = 2.000

AS our statistic is bigger than the critical value we **reject Ho**

In [ ]:
# Using python to find the p value and critical value

from scipy.stats import t

print("P value is: ", 1- t.cdf(statistic,n1+n2-2))
print("Critical Value of z is: ", t.ppf(0.025, n1+n2-2)) #alpha is 0.05

P value is:  0.0005783712704484634
Critical Value of z is:  -1.9996235841149783


If you look at the p-value will 